File: PrimeMinistersByJava/primeministers/Reader.java
package primeministers;

import java.io.File;
import java.util.ArrayList;
import java.util.List;
import java.util.Arrays;
import java.util.stream.Stream;

import libraries.Condition;
import libraries.ImageUtil;

public class Reader extends IO {

	private String aClassName;

	private String filename;

	/**
	* リーダのコンストラクタ
	*/
	public Reader(final Table aTable, final File aCSVFilePath, Object aClassName) {
		super(aTable);
		super.setCSVFilePath(aCSVFilePath);
		this.aClassName = aClassName.toString();
		return;
	}

	public List<String> organize(List<String> beforeList) {
		return null;
	}

	/**
	* オブジェクトを指定文字で分割する。ついでにファイルをフルパスにする
	* @return List<String> 分割後の文字列
	*/
	private List<String> split(String aString, String aSeparator) {
		List<String> result = new ArrayList<>();
		Arrays.asList(aString.split(aSeparator)).forEach(aSplittedWord -> {
			new Condition(() -> ImageUtil.isImageFile(aSplittedWord))
			.ifThenElse(() -> {
				result.add(aSplittedWord);
			},
			() -> {
				result.add(aSplittedWord);
			});
		});
		return result;
	}

	/**
	* CSVファイルの読み込み処理
	*/
	public List<List<String>> perform() {
		// 保存したCSVファイルを読み込む
		final List<String> aListOfCSV = super.readTextFromFile(super.csvFilePath());
		// csvをカンマごとに区切り、List<List<String>>とする
		List<List<String>> splittedList = new ArrayList<>();
		aListOfCSV.forEach(aLine -> {
			splittedList.add(this.split(aLine, ","));
		});
		return splittedList;
	}

}

File: PrimeMinistersByPython/primeministers/reader.py
from .io import IO

class Reader(IO):
	def __init__(self, a_table, a_csv_filepath, a_classname):
		super(Reader, self).__init__(a_table)
		self.set_csv_filepath(a_csv_filepath)
		self.a_classname = a_classname
		return

	def perform(self):
		# print(self.read_text_from_file(self.get_csv_filepath()))
		# for a_line in self.read_text_from_file(self.get_csv_filepath()):
		# 	print(a_line.split(','))
		return [ a_line.split(',') for a_line in self.read_text_from_file(self.get_csv_filepath())]